### PR TITLE
Issue #605 Add libvirt version check as part of preflight

### PR DESCRIPTION
**Pull request by praveenkumar**

### CODE REVIEW

**Comment by zeenix** [comment]
Patch looks good to me.

File: pkg/crc/preflight/preflight_linux.go [hunk 1]
```diff
@@ -47,6 +47,11 @@ func StartPreflightChecks(vmDriver string) {
 		"Checking if libvirt daemon is running",
 		config.GetBool(cmdConfig.WarnCheckLibvirtRunning.Name),
 	)
+	preflightCheckSucceedsOrFails(config.GetBool(cmdConfig.SkipCheckLibvirtVersionCheck.Name),
+		checkLibvirtVersion,
+		"Checking if supported libvirt version installed",
```
Comment (cfergeau) [inline pkg/crc/preflight/preflight_linux.go]: "if a supported libvirt version is installed"

File: pkg/crc/preflight/preflight_checks_linux.go [hunk 2]
```diff
@@ -146,6 +149,28 @@ func fixLibvirtEnabled() (bool, error) {
...
+	if err != nil {
+		return false, fmt.Errorf("Unable to parse installed libvirt version %v", err)
+	}
+	supportedLibvirtVersion, err := semver.NewVersion(minSupportedLibvirtVersion)
+	if err != nil {
+		return false, fmt.Errorf("Unable to parse %s libvirt version %v", minSupportedLibvirtVersion, err)
+	}
+
+	if installedLibvirtVersion.LessThan(supportedLibvirtVersion) {
+		return false, fmt.Errorf("You have %s version of libvirt installed but %s or higher required", installedLibvirtVersion.String(), minSupportedLibvirtVersion)
```
Comment (cfergeau) [inline pkg/crc/preflight/preflight_checks_linux.go]: "libvirt version %s is installed, but %s or higher is required" ?

File: pkg/crc/preflight/preflight_checks_linux.go [hunk 3]
```diff
@@ -28,6 +30,7 @@ const (
 	libvirtDriverVersion        = "0.12.5"
 	crcDnsmasqConfigFile        = "crc.conf"
 	crcNetworkManagerConfigFile = "crc-nm-dnsmasq.conf"
+	minSupportedLibvirtVersion  = "3.4.0"
```
Comment (cfergeau) [inline pkg/crc/preflight/preflight_checks_linux.go]: Add a comment that this should be kept in sync with code-ready/machine-driver-libvirt go.mod?

Comment (praveenkumar) [reply pkg/crc/preflight/preflight_checks_linux.go]: @cfergeau done

File: pkg/crc/preflight/preflight_checks_linux.go (hunk 2 above)
Comment (praveenkumar) [reply pkg/crc/preflight/preflight_checks_linux.go]: @cfergeau updated

File: go.mod [hunk 4]
```diff
@@ -3,6 +3,7 @@ module github.com/code-ready/crc
 require (
 	github.com/BurntSushi/toml v0.3.1 // indirect
 	github.com/DATA-DOG/godog v0.7.13
+	github.com/Masterminds/semver v1.5.0
```
Comment (cfergeau) [inline go.mod]: Wouldn't https://github.com/coreos/go-semver make more sense for us?

Comment (praveenkumar) [reply go.mod]: @cfergeau that is outdated one, no development from last 2 years :)

**Pull request merged**